ios/Classes/internal/core/PowerAuthServiceRegistry.swift

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2025 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import Flutter
+import PowerAuth2
+import PowerAuthCore
+
+/// Singleton registry for PowerAuth services.
+///
+/// This registry is responsible for managing the lifecycle of PowerAuth services and
+/// providing access to them.
+///
+/// There is a single instance of this registry, and it is created
+/// when the first plugin is attached. This is to prevent plugins (each isolate can have its own instance)
+/// from creating multiple instances of the registry, which could lead to conflicts and unexpected
+/// PowerAuth states.
+internal class PowerAuthServiceRegistry {
+
+    // MARK: - Singleton access
+
+    private init() {}
+
+    private static var _shared: PowerAuthServiceRegistry?
+
+    private static let _lock = Lock()
+
+    internal static var shared: PowerAuthServiceRegistry {
```

Comment:
This whole thing could be just:

```swift
internal static let shared = PowerAuthServiceRegistry()
```

We're guarding instance creation here, but static properties in Swift are lazy and thread-safe.

The backing field could be completly removed then.

Reply:
On another look, I think we don't need to create singleton instance at all, and all the instance methods/properties could be static anyway.